src/3d/mesh/qgsmeshterraingenerator.cpp

```diff
@@ -122,7 +122,7 @@ QgsTerrainGenerator *QgsMeshTerrainGenerator::clone() const
 
 QgsTerrainGenerator::Type QgsMeshTerrainGenerator::type() const {return QgsTerrainGenerator::Mesh;}
 
-QgsRectangle QgsMeshTerrainGenerator::extent() const
+QgsRectangle QgsMeshTerrainGenerator::rootChunkExtent() const
```

Comment:
Why using 2D QgsRectangle instead of QgsAABB for 3D objects? This 2D extent won't work with 4978 projections.

This is not a remark specific to your work but rather a general observation on QGIS 3D and clipping (3D dimension is striped to 2D).

Reply:
In general, QgsRectangle is used for _map_ coordinates and QgsAABB for _world_ coordinates. This method returns the terrain generator's root chunk extent in _map coordinates_ so a QgsRectangle is used. The terrain generator logic is built upon projected coordinate systems and while 4978 is technically projected, it is geocentric and may not play well with the 3d view depending on lat-lon